### PR TITLE
Java 8 & Retrolambda

### PR DESCRIPTION
**Pull request by psh**
Changes made in this PR

* Added the _retrolambda_ gradle plugin and updated the language level to `JavaVersion.VERSION_1_8`
* Introduced lambdas + method references as suggested by Android Studio
* Caught a couple of other suggestions from code analysis over the language level

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/contributions/UploadCountClient.java [hunk 1]
```diff
@@ -14,10 +14,10 @@
 import fr.free.nrw.commons.concurrency.ThreadPoolExecutorService;
 import timber.log.Timber;
 
-public class UploadCountClient {
+class UploadCountClient {
     private ThreadPoolExecutorService threadPoolExecutor;
 
-    public UploadCountClient() {
+    UploadCountClient() {
```
Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/contributions/UploadCountClient.java]: Sorry if I missed something, but is there a reason why this class and its constructor was made non-public?

**Comment by misaochan** [comment]
Aside from the above question, it looks good to me and works well. :) The category hang bug #667  did occur to me while testing, but that is unrelated to this PR I believe. I will merge soon if no one else has any issues with this.

File: app/src/main/java/fr/free/nrw/commons/contributions/UploadCountClient.java (hunk 1 above)
Comment (psh) [inline app/src/main/java/fr/free/nrw/commons/contributions/UploadCountClient.java]: There's only a single usage of `UploadCountClient` - line 271 of `ContributionsActivity` - and that is in the same package so I tightened up the visibility of the class + constructor.  Happy to reverse the change if you prefer.

Comment (misaochan) [inline app/src/main/java/fr/free/nrw/commons/contributions/UploadCountClient.java]: Fair point, this is a fairly new class after all and I don't think any other package is likely to use it. I'm good with the change in that case. :)

**Comment by misaochan** [comment]
@psh , thanks for the great work so far ! We'd like to add your name to the CREDITS.md, if you're okay with that. :) If so, what name would you prefer us to use?

**Pull request merged**